File: cpp/Validate-Stack-Sequences/Validate-Stack-Sequences.cpp
class Solution {
public:
    bool validateStackSequences(vector<int>& pushed, vector<int>& popped) {
        int n = pushed.size();
        int m = popped.size();
        if (n!=m) return false;
        stack<int> s;
        int index = 0;
        for (int i=0;i<n;i++)
        {
            s.push(pushed[i]);
            while(!s.empty() && s.top()==popped[index])
            {
                s.pop();
                index++;
            }
        }
        return s.empty();
    }
};

File: python/Validate-Stack-Sequences/Validate-Stack-Sequences.py
class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:
        n = len(pushed)
        m = len(popped)
        if n!=m:
            return False
        s = []
        index = 0
        for i in range(n):
            s.append(pushed[i])
            while len(s)>0 and s[-1]==popped[index]:
                s.pop()
                index+=1
        return len(s)==0
